src/components/ChatEntry.js

```diff
@@ -1,22 +1,38 @@
 import React from 'react';
+import { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const [color, setColor] = useState(false);
+
+  const handleClick = () => {
+    if (color === true) {
+      setColor(false)
+      props.decrementLikes()
+    } else if (color === false) {
+      setColor(true)
+      props.incrementLikes()
+    }
+  }
+ 
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time={props.timeStamp}/></p>
+        <button className="like" onClick={handleClick}>{!color ? '🤍' : '❤️'}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
```

Comment:
These are the basic data we need to display a message without interaction, but the message data also includes an `id` and `liked` value. We should list those in our prop types, and since they are even included in the Wave 1 and 2 tests, we could safely mark them as required.
```js
  id: PropTypes.number.isRequired,
  liked: PropTypes.bool.isRequired,
```

For any callbacks introduced for wave 3, we should still list them in the prop types, but we can leave `isRequired` off to avoid test warnings
```js
  // the current callbacks
  incrementLikes: PropTypes.func,
  decrementLikes: PropTypes.func,
  // preferred callback (single callback we pass the id of a clicked message to)
  onLikeClicked: PropTypes.func,
```
While leaving these not marked `isRequired` avoids a bunch of warnings when running the tests (the tests don't know what you're going to call the event handlers), we'd also like to include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (props.onLikeClicked) {
        props.onLikeClicked(props.id);
    }
```
which would only try to call the function if it's not falsy (and `undefined` is falsy).